src/main/c/Objects/pyjtype.c

```diff
@@ -271,6 +282,28 @@ static int addMethods(JNIEnv* env, PyObject* dict, jclass clazz)
             return -1;
         }
 
+        if (functionalInterface == JNI_TRUE) {
+            jint modifiers = java_lang_reflect_Member_getModifiers(env, rmethod);
+            jboolean isAbstract;
+            if (process_java_exception(env)) {
+                Py_DECREF(pymethod);
+                return -1;
+            }
+            isAbstract = java_lang_reflect_Modifier_isAbstract(env, modifiers);
+            if (process_java_exception(env)) {
+                Py_DECREF(pymethod);
+                return -1;
+            }
+            if (isAbstract) {
+                if (oneAbstractPyJMethod) {
```

Comment:
This if statement could use a comment, that it is checking if we had already had a method, in which case there were two methods and therefore it is not a functional interface.  It took me a bit to figure out how this was working, hence why I'd like a comment.

Reply:
I added a comment. I was a little disappointed to be adding the extra complexity into this function, it is already doing a few different things I would have preferred to keep the functional interface more separate but I couldn't figure out how to do that without duplicating a bunch of this code and making the problem much harder so I opted to leave it mixed in with the other method creation. Anyway, comments are hopefully going to keep the complexity manageable so I appreciate you taking a fresh look at where they are needed.